pkg/startup/v15/startup.go

```diff
@@ -251,3 +254,55 @@ func (s *startup) writeFiles(role api.AgentPoolProfileRole, w writers.Writer, ho
 
 	return w.Close()
 }
+
+// WriteSearchDomain queries for the search domain and writes to
+// /etc/dhcp/dhclient-eth0.conf.  This is for private api clusters only
+func (s *startup) WriteSearchDomain(ctx context.Context, log *logrus.Entry) error {
+	spp := &s.cs.Properties.WorkerServicePrincipalProfile
+
+	s.log.Info("creating clients")
+	authorizer, err := azureclient.NewAuthorizer(spp.ClientID, spp.Secret, s.cs.Properties.AzProfile.TenantID, "")
+	if err != nil {
+		return err
+	}
+
+	ncli := network.NewInterfacesClient(ctx, log, s.cs.Properties.AzProfile.SubscriptionID, authorizer)
+	var dnsString string
+	nic, err := ncli.GetVirtualMachineScaleSetNetworkInterface(ctx, s.cs.Properties.AzProfile.ResourceGroup, "ss-master", "0", "nic", "")
+	if err != nil {
+		return err
+	}
+	if nic.DNSSettings != nil {
+		dnsString = to.String(nic.DNSSettings.InternalDomainNameSuffix)
+	}
+	s.log.Info("writing custom dns")
+
+	domainSettings := []string{
+		"interface \"eth0\" {\n",
+		fmt.Sprintf("    supersede domain-name \"%s\";\n", dnsString),
+		fmt.Sprintf("    supersede domain-search \"%s\";\n", dnsString),
+		"}\n",
+	}
+
+	// write to dnsmasq.conf and see if that is sufficient
```

Comment:
is this a TODO or an old comment, somewhat confusing

Reply:
removed